File: src/Entity/Poste.php
<?php

namespace App\Entity;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use App\Repository\PosteRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Entity\User;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass : PosteRepository::class)]
class Poste
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    
    private ?int $id=null;
    
    #[Assert\NotBlank(message : "veuillez entrez Un titre valides")]
    #[ORM\Column(length : 200)]
    private ?string $titre=null;
    #[Assert\NotBlank(message : "veuillez entrez une description valides")]
    #[ORM\Column(length : 500)]
    private ?string $description=null;

    #[ORM\Column(length : 250)]
    private ?string $img=null;
    #[Assert\NotBlank(message : "veuillez choisir un domaine")]
    #[ORM\Column(length : 250)]
    private ?string $domaine=null;

    #[ORM\Column(length: '250')]
    private ?string $date=null;

    #[ORM\ManyToOne(targetEntity:User::class)]
    #[ORM\JoinColumn(name: 'id_user', referencedColumnName: 'id')]
    private ?User $idUser=null;
    
    #[ORM\OneToMany(targetEntity:Likez::class, mappedBy : "idPoste", cascade :["persist", "remove"])]
   
    private  $likes=null;
    #[ORM\OneToMany(targetEntity:Report::class, mappedBy : "idPoste", cascade :["persist", "remove"])]
   
    private  $reports=null;

    public function __construct()
    {
        $this->likes = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): self
    {
        $this->titre = $titre;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getImg(): ?string
    {
        return $this->img;
    }

    public function setImg(string $img): self
    {
        $this->img = $img;

        return $this;
    }

    public function getDomaine(): ?string
    {
        return $this->domaine;
    }

    public function setDomaine(string $domaine): self
    {
        $this->domaine = $domaine;

        return $this;
    }

    public function getDate(): ?string
    {
        return $this->date;
    }

    public function setDate(string $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getIdUser(): ?User
    {
        return $this->idUser;
    }

    public function setIdUser(?User $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }
/**
     * @return Collection|Likez[]
     */
    public function getLikes(): Collection
    {
        return $this->likes;
    }
    /**
    * @return Collection|Report[]
     */
    public function getReports(): Collection
    {
        return $this->reports;
    }

    public function addLike(Likez $like): self
    {
        if (!$this->likes->contains($like)) {
            $this->likes[] = $like;
            $like->setPost($this);
        }

        return $this;
    }

    public function removeLike(Likez $like): self
    {
        if ($this->likes->contains($like)) {
            $this->likes->removeElement($like);
            // set the owning side to null (unless already changed)
            if ($like->getPost() === $this) {
                $like->setPost(null);
            }
        }

        return $this;
    }

    public function getTotalLikes(): int
    {
        return count($this->likes);
    }

    public function isLikeByUser(User $Client):bool{
        foreach ($this->likes as $Like){
            if ($Like->getIdUser() === $Client)return true;
        }
        return false;
    }
}

File: job4u-publication/job4u/src/com/job4u/entities/Poste.java
package com.job4u.entities;

import java.util.Date;
import com.job4u.utils.*;

public class Poste {

    private int id;
    private User user;
    private String titre;
    private String description;
    private String image;
    private String domaine;
    private Date date;

    public Poste() {}

    public Poste(int id, User user, String titre, String description, String image, String domaine, Date date) {
        this.id = id;
        this.user = user;
        this.titre = titre;
        this.description = description;
        this.image = image;
        this.domaine = domaine;
        this.date = date;
    }

    public Poste(User user, String titre, String description, String image, String domaine, Date date) {
        this.user = user;
        this.titre = titre;
        this.description = description;
        this.image = image;
        this.domaine = domaine;
        this.date = date;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getDomaine() {
        return domaine;
    }

    public void setDomaine(String domaine) {
        this.domaine = domaine;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    @Override
    public String toString() {
        return titre;
    }

}